src/geode/geosciences/lso_input.cpp

```diff
@@ -471,6 +494,102 @@ namespace
                 builder->create_tetrahedron( vertices );
             }
             builder->compute_polyhedron_adjacencies();
+
+            create_block_vertices_attributes(
+                block_id, inverse_vertex_mapping );
+        }
+
+        void create_block_vertices_attributes( const geode::uuid& block_id,
+            absl::Span< const geode::index_t > inverse_vertex_mapping )
+        {
+            const auto& block_mesh = model_.block( block_id ).mesh();
+            for( const auto attr_id : geode::Indices{ prop_header_.names } )
+            {
+                const auto nb_attribute_items = prop_header_.esizes[attr_id];
+                if( nb_attribute_items == 1 )
+                {
+                    auto attribute = block_mesh.vertex_attribute_manager()
+                                         .template find_or_create_attribute<
+                                             geode::VariableAttribute, double >(
+                                             prop_header_.names[attr_id], 0 );
+                    for( const auto pt_id :
+                        geode::Range{ block_mesh.nb_vertices() } )
+                    {
+                        attribute->set_value( pt_id,
+                            vertices_attributes_
+                                [attr_id][inverse_vertex_mapping[pt_id]] );
+                    }
+                }
+                else if( nb_attribute_items == 2 )
+                {
+                    std::array< double, 2 > value_array;
+                    auto attribute =
+                        block_mesh.vertex_attribute_manager()
+                            .template find_or_create_attribute<
+                                geode::VariableAttribute,
+                                std::array< double, 2 > >(
+                                prop_header_.names[attr_id], value_array );
+                    for( const auto pt_id :
+                        geode::Range{ block_mesh.nb_vertices() } )
+                    {
+                        for( const auto item_id :
+                            geode::Range{ nb_attribute_items } )
+                        {
+                            value_array[item_id] = vertices_attributes_
+                                [attr_id][inverse_vertex_mapping[pt_id]
+                                              * nb_attribute_items
+                                          + item_id];
+                        }
+                        attribute->set_value( pt_id, value_array );
+                    }
```

Comment:
all of this could be a templated method to reuse for each dimension

Reply:
or event better, template the entire container type, so it could also be used for the last case with vector